### PR TITLE
Algo change to 192_7 from dev work done in May 2018

### PR DESCRIPTION
**Pull request by mattpass**
This PR is from work done in May 2018 before the the rebase idea was started upon. The status of it according to @michaelotis is that:

- It was tested only a little, not a whole lot, just a little. It was producing blocks just fine but it was only ever attempted on a new chain and not as a fork.
- It's believed that, similar to other coins, there may be some re-targeting issues that may need to be resolved

It should be noted that a few commits on the BTCPrivate/BitcoinPrivate repo have happened since and not in the repo being proposed to be merged in, so worthwhile considering especially re any potential missed areas or conflicts. The commits are from this point to master's head: https://github.com/BTCPrivate/BitcoinPrivate/compare/4045199486c8182500572447b659209b5d274994...master

Proposed steps to take are therefore:
- [x] Check over commits in this PR
- [x] Fix merge conflicts
- [x] Check over the commits not in that PR in the commit range link above for anything potentially missing or would cause conflicts
- [x] Test it does indeed run perfectly fine as a new chain
- [x] Test it does also run perfectly as a fork
- [x] Consider the re-targeting issues mentioned
- [x] Test again thoroughly in a small testnet as a final check

Extra steps to take and comments on the above welcomed.

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -198,17 +200,16 @@ class CTestNetParams : public CMainParams {
         nPruneAfterHeight = 1000;
 
         //! Modify the testnet genesis block so the timestamp is valid for a later start.
-        genesis.nTime = 1479443947;
+        genesis.nTime = 1527600786;
```
Comment (dskjkljdas) [inline src/chainparams.cpp]: does this need to be redone?

File: src/chainparams.cpp [hunk 2]
```diff
@@ -152,6 +154,10 @@ class CMainParams : public CChainParams {
 
         nForkStartHeight = 272991;
         nForkHeightRange = 5467;
+
+        nEquihashForkHeight = 600001;
+        nEquihashNnew = 144;
```
Comment (dskjkljdas) [inline src/chainparams.cpp]: 197?

File: src/chainparams.cpp [hunk 3]
```diff
@@ -152,6 +154,10 @@ class CMainParams : public CChainParams {
...
+        nEquihashKnew = 5;
```
Comment (dskjkljdas) [inline src/chainparams.cpp]: 2?

**Comment by bigdaveakers** [comment]
Are the changes in this 'similar' to those in #181? Can #181 be closed in favour of this change?

**Comment by bigdaveakers** [comment]
I am not comfortable with block 600001, I would favour earlier implementation if the desire is to enhance 'ASIC' resistance. I would suspect 7 months is plenty of time to develop a hardware solution meaning that we continually chase our tail to eliminate the threat.

**Comment by dskjkljdas** [comment]
> I am not comfortable with block 600001, I would favour earlier implementation if the desire is to enhance 'ASIC' resistance. I would suspect 7 months is plenty of time to develop a hardware solution meaning that we continually chase our tail to eliminate the threat.

Agreed, the difficulty bomb can be reprogrammed to 1 block after the alogo change fork blockheight

**Comment by dskjkljdas** [comment]
> Are the changes in this 'similar' to those in #181? Can #181 be closed in favour of this change?

Yeah the base code is the same, 202 has all the shared contributions 181 seems to be the initial guide

**Comment by bigdaveakers** [comment]
> Agreed, the difficulty bomb can be reprogrammed to 1 block after the alogo change fork blockheight

1 may be a bit controversial, but it would be effective.

**Comment by interbiznw** [comment]
Testing now @ https://testnet2.btcprivate.org/ If anyone would like to join in with some hashrate so we can analyze the difficulty adjustment thoroughly. TYIA

**Comment by interbiznw** [comment]
Still in testing phase on TESTNET @ https://testnet2.btcprivate.org/ If anyone would like to join in with some hashrate so we can analyze the difficulty adjustment thoroughly. TYIA

**Pull request merged**